src/ECER.Resources.Documents/Applications/ApplicationRepository.cs

```diff
@@ -237,10 +237,13 @@ private async Task<string> SubmitCharacterReference(string characterReferenceId,
     var characterReference = context.ecer_CharacterReferenceSet.Single(c => c.ecer_CharacterReferenceId == Guid.Parse(characterReferenceId));
 
     mapper.Map(request, characterReference);
-    var province = context.ecer_ProvinceSet.SingleOrDefault(p => p.ecer_ProvinceId == Guid.Parse(request.ReferenceContactInformation.CertificateProvinceId));
-    if (province != null)
+    if (!string.IsNullOrWhiteSpace(request.ReferenceContactInformation.CertificateProvinceId))
     {
-      context.AddLink(characterReference, ecer_CharacterReference.Fields.ecer_characterreference_RefCertifiedProvinceId, province);
+      var province = context.ecer_ProvinceSet.SingleOrDefault(p => p.ecer_ProvinceId == Guid.Parse(request.ReferenceContactInformation.CertificateProvinceId));
```

Comment:
What happens when this is not a guid? is province mandatory?

Reply:
Province is optional, so Province Id can possibly have following value types -> empty, null or a GUID.
Added Guid try parse check.